src/index.rs

```diff
@@ -112,6 +119,14 @@ impl Index {
         &self.chain
     }
 
+    pub(crate) fn limit_result<T>(&self, entries: impl Iterator<Item = T>) -> Result<Vec<T>> {
+        let entries: Vec<T> = entries.collect();
```

Comment:
I guess it'd be better to use `entries.by_ref().take(self.lookup_limit).collect()` and then check `entries.next().is_some()`? In that case the function should also take `impl Iterator<Item = T> + FusedIterator`.

Reply:
Sounds good, updated the code in https://github.com/romanz/electrs/pull/458/commits/31765e3e49ea93c97721675e2c06e2a9efa76812.
I have called `.fuse()` [here](https://github.com/romanz/electrs/pull/458/commits/31765e3e49ea93c97721675e2c06e2a9efa76812#diff-701633ed548410624cfcae90d38a986adbd60deb5aeedd92355fe2e7702a9c18R123) following the suggestion from https://doc.rust-lang.org/std/iter/trait.FusedIterator.html.